File: 7kyu/7kyu - Sum of all the multiples of 3 or 5.js
// 7kyu - Sum of all the multiples of 3 or 5


// const findSum = n => Array(n).fill().map((_, i) => i + 1).filter(v => v % 3 == 0 || v % 5 == 0).reduce((a, b) => a + b)
// const findSum = n => Array(n).fill().map((_, i) => i + 1).filter(v => !(v % 3 && v % 5)).reduce((a, b) => a + b)
// const findSum = n => Array(n).fill().map((_, i) => i + 1).reduce((acc, cur, i, a) => !(cur % 3 && cur % 5) ? acc += cur : acc, 0)
// const findSum = n => [...Array(n + 1).keys()].filter(cur => cur % 3 === 0 || cur % 5 === 0).reduce((acc, cur) => acc + cur, 0);
// const findSum = n => [...Array(n + 1).keys()].filter(v => !(v % 3 && v % 5)).reduce((a, b) => a + b)
// const findSum = n => [...Array(n)].map((_, i) => ++i).reduce((r, e) => r + (e % 5 && e % 3 ? 0 : e), 0)
const findSum = n => [...Array(n+1).keys()].reduce((a, b) => a + (b % 3 && b % 5 ? 0 : b))


// function findSum(n) {
//     for (var sum = 0, i = 0; i <=n; i++)
//         if (i % 3 == 0 || i % 5 == 0)
//             sum += i
//     return sum
// }

q = findSum(5) // 8
q
q = findSum(10) // 33
q

File: 7kyu/7kyu - Sum of all the multiples of 3 or 5.py
# 7kyu - Sum of all the multiples of 3 or 5

"""   Write function find

Upto and including n, this function will return the sum of all multiples of 3 and 5.

find(5) should return 8 (3 + 5)
find(10) should return 33 (3 + 5 + 6 + 9 + 10) """


def find(n):
    return sum(x for x in range(1, n+1) if x % 3 == 0 or x % 5 == 0)


q = find(5)  # 8
q
q = find(10)  # 33
q
